File: src/main/java/io/stouder/adonis/AdonisBundle.java
package io.stouder.adonis;

import com.intellij.DynamicBundle;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;

import java.util.Arrays;

public class AdonisBundle extends DynamicBundle {
    public static final AdonisBundle INSTANCE = new AdonisBundle();

    private AdonisBundle() {
        super("messages.AdonisBundle");
    }

    public static String message(@PropertyKey(resourceBundle = "messages.AdonisBundle") @NotNull String key, @NotNull Object... params) {
        return INSTANCE.getMessage(key, Arrays.copyOf(params, params.length));
    }
}

File: src/main/java/io/stouder/adonis/AdonisBundle.kt
package io.stouder.adonis

import com.intellij.DynamicBundle
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.PropertyKey

object AdonisBundle : DynamicBundle("messages.AdonisBundle") {
    @JvmStatic
    fun message(
        @PropertyKey(resourceBundle = "messages.AdonisBundle") key: @NotNull String, vararg params: @NotNull Any): String {
        return getMessage(key, *params)
    }
}